src/company_cmd.cpp

```diff
@@ -108,11 +108,11 @@ void SetLocalCompany(CompanyID new_company)
 	InvalidateWindowData(WC_SEND_NETWORK_MSG, DESTTYPE_TEAM, _local_company);
 
 	assert(IsLocalCompany());
-
+	
 	_current_company = _local_company = new_company;
 
-	/* Delete any construction windows... */
-	DeleteConstructionWindows();
+	/* Recreate company windows */
+	InvalidateCompanyWindows(Company::GetIfValid(new_company));
```

Comment:
`Company::GetIfValid(new_company)` can return `nullptr`, causing `company->index` to crash in `InvalidateCompanyWindow()`

Reply:
Can Company::Get() return a `nullptr`?